src/plugins/teleport-plugin-react-base-component/utils.ts

```diff
@@ -293,20 +289,29 @@ const makeStateHookAST = (stateIdentifier: StateIdentifier, t = types) => {
 }
 
 const makeRepeatStructureWithMap = (
-  dataSource: string | any[],
+  dataSource: UIDLNodeAttributeValue,
   content: types.JSXElement,
   meta: Record<string, any> = {},
   t = types
 ) => {
   const iteratorName = meta.iteratorName || 'item'
   const keyIdentifier = meta.useIndex ? 'index' : iteratorName
 
-  const source =
-    typeof dataSource === 'string'
-      ? t.identifier(dataSource)
-      : t.arrayExpression(dataSource.map((element) => convertValueToLiteral(element)))
+  let source
+
+  if (dataSource.type === 'static') {
```

Comment:
I suggest we switch by type and throw on default when encountering unknown types. Helps a lot with debugging and TDD.